hotel-api-persistence/src/main/java/com/fponce/hotelapp/persistence/impl/HotelRepositoryImpl.java

```diff
@@ -30,4 +45,23 @@ public void createHotel(UUID id, String name, int category) throws SQLException
         sqlWithParameters.apply(preparedStatement);
         preparedStatement.executeUpdate();
     }
+
+    /**
+     * Returns the list of Hotels registered in the database. It can use a name
+     * to filter the search.
+     *
+     * @return
+     */
+    @Override
+    public List<Hotel> getHotels() throws SQLException {
+        Connection connection = dataSource.getConnection();
+        String getHotelsQuery = "SELECT * FROM HOTEL";
+        Statement statement = connection.createStatement();
+        ResultSet resultSet = statement.executeQuery(getHotelsQuery);
+        Stream<ResultSet> stream = StreamSupport.stream(Spliterators.spliteratorUnknownSize(new ResultSetIterator(resultSet), 0), false);
```

Comment:
Why is StreamSupport and Spliterator needed?

Reply:
I used StreamSupport to convert the Spliterator into Stream, since ResultSet can't be used as stream, but now I think it's adding unnecessary complexity just to convert ResultSets into Streams.

Reply:
I am removing this, and implementing plain ResultSet iteration.